docs/application/dotnet/guides/app-management/component-port.md

```diff
@@ -0,0 +1,467 @@
+                base.OnStartCommand(appControl, restarted);
+            }
+
+            public override void OnDestroy()
+            {
+                Log.Info("OnDestroy()");
+                base.OnDestroy();
+
+                if (_port != null)
+                {
+                    _port.RequestReceived -= OnRequestReceived;
+                    _port.Dispose();
+                    _port = null;
+                }
+            }
+        }
+    }
+    ```
+
+3.  Set up the receiving thread of the component.
+
+- To have the receiving thread of the component wait for incoming requests, call `WaitForEvent()` of the `Tizen.Applications.ComponentBased.ComponentPort` class.
+- If `WaitForEvent()` is called in the main thread, then `WaitForEvent()` can not be called until the `Cancel()` is called. 
+- `WaitForEvent()` starts the main loop to wait for events from other components. To avoid blocking the main thread, it is recommended to use the `Tizen.Applications.ComponentBased.ComponentTask` class.
+- To handle the access control, call `AddPrivilege()` of the `Tizen.Applications.ComponentBased.ComponentPort` class. If a client does not have permission, the request is rejected with the permission denied error.
+- To handle the received request, define and register an event handler for the `RequestReceived` event of the `Tizen.Applications.ComponentBased.ComponentPort` class as follows:
+
+    ```
+    {
+        _port = new ComponentPort("ServerService");
+        _port.RequestReceived += OnRequestReceived;
+        _port.AddPrivilege("http://tizen.org/privilege/datasharing");
+        _thread = new Thread(new ThreadStart(OnThread));
+        _thread.Start();
+    }
+
+    private void OnThread()
+    {
+        Log.Info("OnTrhead()");
+        _port.WaitForEvent();
+    }
+
+    private void OnRequestedReceived(object sender, RequestEventArgs args)
+    {
+        Log.Info("Sender: + args.Sender);
+        var request = args.Request;
+        if (request.GetType() == typeof(Request))
+        {
+            var req = (Request)request;
+            Log.Info("Command: " + req.Command);
+            Log.Info("Sequence: " + req.Sequence);
+            Log.Info("Message: " + req.Message);
+
+            if (args.IsReplyRequested)
+            {
+                args.Reply = new Response(req.Command, req.Sequence, 0);
+            }
+        }
+    }
+    ```
+
+4.  Send the request in the sending component as follows:
+
+    1. Call `WaitForEvent()` of the `Tizen.Applications.ComponentBased.ComponentPort` class in the thread, if you want to receive the reply from `ServerService.Tizen`.
+    2. Call `WaitForPort`() of the `Tizen.Applications.ComponentBased.ComponentPort` class to wait until the target port is ready.
+    3. Use `Send()` of the `Tizen.Applications.ComponentBased.ComponentPort` class to send the request.
+    4. Provide the request to be sent as an instance of the `ClientService.Tizen.Request` class as follows:
+
+    ```
```

Comment:
Kindly align the code segment inline to point 4.

Reply:
Done